<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1280</width>
    <height>720</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>1280</width>
    <height>720</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>1280</width>
    <height>720</height>
   </size>
  </property>
  <property name="font">
   <font>
    <pointsize>12</pointsize>
   </font>
  </property>
  <property name="windowTitle">
   <string>std::set &amp; std::map &amp; std::unordered_map</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QGroupBox" name="groupBox">
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="title">
       <string>std::set&lt;KeyType&gt;</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <widget class="QTreeWidget" name="set_view">
       <property name="geometry">
        <rect>
         <x>7</x>
         <y>40</y>
         <width>400</width>
         <height>400</height>
        </rect>
       </property>
       <column>
        <property name="text">
         <string>set</string>
        </property>
       </column>
      </widget>
      <widget class="QGroupBox" name="groupBox_6">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>440</y>
         <width>391</width>
         <height>251</height>
        </rect>
       </property>
       <property name="title">
        <string>Действия</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_4">
        <item row="0" column="1">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Key</string>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="Insert_set">
          <property name="text">
           <string>Insert</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="insert_set_key"/>
        </item>
        <item row="2" column="0" colspan="2">
         <widget class="QPushButton" name="delete_set">
          <property name="text">
           <string>Delete</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="2">
         <widget class="QPushButton" name="show_set">
          <property name="text">
           <string>Show</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="0" column="2">
     <widget class="QGroupBox" name="groupBox_3">
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="title">
       <string>std::unordered_map&lt;KeyType, ValueType&gt;</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <widget class="QTableWidget" name="unordered_map_view">
       <property name="geometry">
        <rect>
         <x>7</x>
         <y>40</y>
         <width>400</width>
         <height>400</height>
        </rect>
       </property>
       <property name="rowCount">
        <number>100</number>
       </property>
       <attribute name="horizontalHeaderMinimumSectionSize">
        <number>55</number>
       </attribute>
       <attribute name="horizontalHeaderDefaultSectionSize">
        <number>200</number>
       </attribute>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <row/>
       <column>
        <property name="text">
         <string>Key</string>
        </property>
       </column>
       <column>
        <property name="text">
         <string>Value</string>
        </property>
       </column>
      </widget>
      <widget class="QGroupBox" name="groupBox_4">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>440</y>
         <width>391</width>
         <height>251</height>
        </rect>
       </property>
       <property name="title">
        <string>Действия</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="1" colspan="2">
         <widget class="QLabel" name="label">
          <property name="font">
           <font>
            <pointsize>10</pointsize>
           </font>
          </property>
          <property name="text">
           <string>Key        Value</string>
          </property>
         </widget>
        </item>
        <item row="1" column="2">
         <widget class="QLineEdit" name="insert_value"/>
        </item>
        <item row="0" column="0">
         <widget class="QPushButton" name="ChangeValue">
          <property name="text">
           <string>Change Value</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QSpinBox" name="insert_key"/>
        </item>
        <item row="1" column="0">
         <widget class="QPushButton" name="GetValue">
          <property name="text">
           <string>GetValue</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QPushButton" name="Contains">
          <property name="text">
           <string>Contains</string>
          </property>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QPushButton" name="Show">
          <property name="text">
           <string>Show</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1" colspan="2">
         <widget class="QPushButton" name="Clear">
          <property name="text">
           <string>Clear(All)</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QPushButton" name="Erase">
          <property name="text">
           <string>Erase</string>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QPushButton" name="Insert">
          <property name="text">
           <string>Insert</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="autoFillBackground">
       <bool>true</bool>
      </property>
      <property name="title">
       <string>std::map&lt;KeyType, ValueType&gt;</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
      <widget class="QTreeWidget" name="map_view">
       <property name="geometry">
        <rect>
         <x>7</x>
         <y>40</y>
         <width>400</width>
         <height>400</height>
        </rect>
       </property>
       <column>
        <property name="text">
         <string notr="true">map</string>
        </property>
       </column>
      </widget>
      <widget class="QGroupBox" name="groupBox_5">
       <property name="geometry">
        <rect>
         <x>10</x>
         <y>440</y>
         <width>391</width>
         <height>256</height>
        </rect>
       </property>
       <property name="title">
        <string>Действия</string>
       </property>
       <layout class="QGridLayout" name="gridLayout_3">
        <item row="1" column="0" colspan="3">
         <widget class="QSpinBox" name="map_key"/>
        </item>
        <item row="5" column="0" colspan="4">
         <widget class="QPushButton" name="Show_map">
          <property name="text">
           <string>Show</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0" colspan="4">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Key                Value</string>
          </property>
         </widget>
        </item>
        <item row="1" column="3">
         <widget class="QLineEdit" name="map_value"/>
        </item>
        <item row="4" column="0" colspan="4">
         <widget class="QPushButton" name="pushButton">
          <property name="text">
           <string>Delete</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0" colspan="4">
         <widget class="QPushButton" name="Insert_map">
          <property name="text">
           <string>Insert</string>
          </property>
         </widget>
        </item>
        <item row="2" column="0" colspan="4">
         <widget class="QPushButton" name="pushButton_2">
          <property name="text">
           <string>At</string>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
